### PR TITLE
fix(react): miscellaneous ecosystem wallet fixes

### PR DESCRIPTION
**Pull request by gregfromstl**
## PR-Codex overview
The focus of this PR is to enhance ecosystem wallet functionality. 

### Detailed summary
- Updated ecosystem wallet connection options
- Improved ecosystem wallet form UI
- Added ecosystem wallet identification logic
- Refactored authentication options to include ecosystem details

> The following files were skipped due to too many changes: `packages/thirdweb/src/react/web/wallets/ecosystem/EcosystemWalletHeader.tsx`, `packages/thirdweb/src/wallets/create-wallet.ts`, `packages/thirdweb/src/wallets/in-app/web/lib/web-connector.ts`, `packages/thirdweb/src/react/web/wallets/shared/OTPLoginUI.tsx`, `packages/thirdweb/src/wallets/in-app/web/lib/auth/passkeys.ts`, `packages/thirdweb/src/wallets/in-app/web/utils/iFrameCommunication/InAppWalletIframeCommunicator.ts`, `packages/thirdweb/src/react/web/wallets/shared/ConnectWalletSocialOptions.tsx`, `packages/thirdweb/src/wallets/in-app/web/lib/in-app-account.ts`, `packages/thirdweb/src/wallets/in-app/core/authentication/index.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by gregfromstl** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3295?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3295** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3295?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3288** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3288?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3281** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3281?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3277** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3277?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3275** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3275?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3270** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3270?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gregfromstl and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/thirdweb/src/wallets/in-app/core/authentication/type.ts [hunk 1]
```diff
@@ -53,8 +42,7 @@ export type SingleStepAuthArgsType =
 
 export type AuthArgsType = (MultiStepAuthArgsType | SingleStepAuthArgsType) & {
   client: ThirdwebClient;
-  walletId: string;
-  partnerId?: string;
+  ecosystemOptions?: EcosystemOptions;
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/in-app/core/authentication/type.ts:45]: ok that's better - how about calling this prop just 'ecosystem' ?

**Pull request merged**